src/models/GitHub.js

```diff
@@ -1,157 +1,158 @@
+        const nextEndCursor = results.search.pageInfo.endCursor
+        return this.getIssuesFromSearch(search, nextEndCursor, allEdges)
+      }
 
-  // https://developer.github.com/v3/activity/notifications/#mark-a-thread-as-read
-  markAsRead(url) {
-    return this.patch(url, { ignoreBody: true })
+      return { tasks: allEdges.map(edge => this.transformEdgeToTask(edge)) }
+    })
   }
 
-  patch(relativeOrAbsoluteUrl, opts) {
-    let url = relativeOrAbsoluteUrl
-    if (url.indexOf('http') !== 0) {
-      url = `${Config.githubApiUrl}/${relativeOrAbsoluteUrl}`
+  graphql(query, variables = {}) {
+    const url = 'https://api.github.com/graphql'
+    const options = {
+      body: JSON.stringify({ query, variables }),
+      method: 'POST',
+      headers: this.getHeaders(),
     }
-    const options = opts || {}
-    options.headers = this.getHeaders()
-    return super.patch(url, options)
+
+    console.log(JSON.stringify(variables))
+
+    return fetch(url, options).then(response => {
+      if (!response.ok) {
+        return response.json().then(json => {
+          throw new Error(`${response.status}: ${JSON.stringify(json)}`)
+        })
+      }
+
+      return response.json().then(result => {
+        if (result.errors) {
+          throw new Error(`GraphQL Error: ${result.errors.map(e => e.message)}`)
+        }
+
+        return result.data
+      })
+    })
   }
 
   getHeaders() {
     if (!this.token) {
       this.token = GitHubAuth.getToken()
     }
     return {
-      Accept: 'application/vnd.github.v3+json',
-      Authorization: `token ${this.token}`,
+      Authorization: `bearer ${this.token}`,
     }
   }
 
-  get(path) {
-    const url = this.getFullUrl(path)
-    const opts = { headers: this.getHeaders() }
-    return super.get(url, opts)
-  }
-
-  getFullUrl(relativeOrAbsoluteUrl) {
-    if (relativeOrAbsoluteUrl.indexOf('http') !== 0) {
-      return `${Config.githubApiUrl}/${relativeOrAbsoluteUrl}`
-    }
-    return relativeOrAbsoluteUrl
-  }
-
-  getNextUrl(headers) {
-    const link = headers.get('Link')
-    if (!link) {
-      return null
+  transformEdgeToTask(edge) {
+    const node = edge.node
+    const type = node.type === 'PullRequest' ? 'pull' : 'issue'
+    return {
+      storageKey: `${type}-${node.id}`,
+      id: node.id,
+      type: node.type,
+      title: node.title,
+      body: node.body,
+      state: node.state,
+      createdAt: node.createdAt,
+      updatedAt: node.updatedAt,
+      isPullRequest: node.type === 'PullRequest',
+      url: node.url,
+      number: node.number,
+      repository: `${node.repository.owner.login}/${node.repository.name}`,
+      repositoryOwner: node.repository.owner.login,
+      repositoryOwnerUrl: node.repository.owner.url,
+      repositoryOwnerAvatar: node.repository.owner.avatarURL,
+      user: node.author.login,
+      userUrl: node.author.url,
+      userAvatar: node.author.avatarURL,
+      comments: node.comments ? node.comments.totalCount : 0,
```

Comment:
Is this preserving the existing behavior where if you're the one who comments on a task, it will resurface in snoozer? I think that's fine, I'd rather introduce GraphQL in this branch and make the fix for not notifying yourself about your own comments in a follow-up branch.